lib/src/keyboard_manager.dart

```diff
@@ -57,21 +60,19 @@ class KeyboardManagerWidgetState extends State<KeyboardManagerWidget> {
   Widget build(BuildContext context) {
     var bottom = MediaQuery.of(context).viewInsets.bottom;
     var keyboardOpen = bottom > 0;
-    var oldKeyboardOpen = _keyboardOpen ?? !keyboardOpen;
+    var oldKeyboardOpen = _keyboardOpen;
     _keyboardOpen = keyboardOpen;
-    
+
     if (_keyboardOpen) {
       dismissed = false;
       _keyboardHeight = bottom;
-      if(!oldKeyboardOpen && activePointer == null) {
-        if(widget.onKeyboardOpen != null)
-          widget.onKeyboardOpen();
+      if (!oldKeyboardOpen && activePointer == null) {
+        if (widget.onKeyboardOpen != null) widget.onKeyboardOpen!();
       }
     } else {
       // Close notification if the keyobard closes while not dragging
-      if(oldKeyboardOpen && activePointer == null) {
-        if(widget.onKeyboardClose != null)
-          widget.onKeyboardClose();
+      if (oldKeyboardOpen && activePointer == null) {
+        if (widget.onKeyboardClose != null) widget.onKeyboardClose!();
```

Comment:
i see both `methodName?.call()` and `methodName!()` used in this PR. is there a reason to use one over another?

Reply:
I've now removed all usages of `methodName!()` to minimise the use of the `!` cast. Incorrectly using the cast causes runtime exceptions like the one fixed by https://github.com/FirebaseExtended/flutterfire/pull/5656